docs/pages/reference/session-recording.mdx

```diff
@@ -0,0 +1,141 @@
+---
+title: Teleport Session Recording
+description: An overview of Teleport's session recording and its configuration
+---
+
+This page provides an overview of how Teleport records sessions and the various
+configuration options that apply to session recordings.
+
+Teleport supports two different recording configurations: synchronous recording
+and asynchronous recording. Additionally, Teleport administrators can configure
+where the recording takes place: at the node or at the proxy.
+
+This results in 4 possible sesssion recording configurations:
+
+- `node-sync`: synchronous recording performed by the SSH node
+- `node`: asynchronous recording performed by the SSH node
```

Comment:
> We're replacing "SSH" with "Server Access", right?

That would be news to me. Though I don't see a harm in sticking with SSH here since it's the most specific.

> This is true for Server Access sessions, but what about others?

Well, that's the thing, SSH is unique in that Teleport software is running at both ends of the connection (Teleport is the SSH server you're connecting to, and Teleport is the SSH client)

For desktop access (as mentioned elsewhere in this PR) there is not a distinction between "node" and "proxy" - the recording is always done on the windows_desktop_service, which is neither a node or proxy. 

Kube and DB access are similar - the Kube cluster or DB that you're connecting to is not Teleport software, it's other software that's totally unaware of Teleport, just like a desktop with RDP.

Reply:
> > We're replacing "SSH" with "Server Access", right?
> 
> That would be news to me. Though I don't see a harm in sticking with SSH here since it's the most specific.

Maybe I have that backwards.

Reply:
> Kube and DB access are similar - the Kube cluster or DB that you're connecting to is not Teleport software, it's other software that's totally unaware of Teleport, just like a desktop with RDP.

You mean similar to Desktop, where the recoding is done at the node running the service, right?

Reply:
So the more accurate statement is that session recording for DB and Kube occurs at the host running the Teleport Service for Database and Kubernetes access?

Reply:
Right. Kube, DB, Desktop (and even App Access) are all alike in this regard, while SSH is a bit unique, which is why I mentioned SSH specifically in the original bullet point that started this thread.